c_glib/arrow-glib/basic-array.cpp

```diff
@@ -544,6 +544,30 @@ garrow_array_view(GArrowArray *array,
   }
 }
 
+/**
+ * garrow_array_diff:
+ * @array: A #GArrowArray.
+ * @other_array: A #GArrowArray to be compared.
+ *
+ * Returns: (transfer full): The string representation of
+ *   the difference between two arrays.
+ *
+ *   It should be freed with g_free() when no longer needed.
+ *
+ * Since: 0.15.0
+ */
+gchar *
+garrow_array_diff(GArrowArray *array, GArrowArray *other_array)
+{
+  const auto arrow_array = garrow_array_get_raw(array);
+  const auto arrow_other_array = garrow_array_get_raw(other_array);
+  std::stringstream diff;
+  arrow_array->Equals(arrow_other_array,
+                      arrow::EqualOptions().diff_sink(&diff));
+  auto string = diff.str();
+  return g_strndup(string.data(), string.size());
```

Comment:
Could you return `NULL` when no difference?

```cpp
if (string.empty()) {
  return NULL;
} else {
  return g_strndup(string.data(), string.size());
}
```

Reply:
I've fixed it.